src/modules/cpu/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,106 @@
+#include "rppdefs.h"
+
+Rpp32f getSquare(Rpp32f &value) 
+{
+  Rpp32f res = value;
+  return (res * res);
+}
+
+Rpp32f getMax(std::vector<float> &values, int length)
+{
+  Rpp32f max = values[0];
+  for (int i = 1; i < length; i++) 
+  {
+    max = std::max(max, values[i]);
+  }
+  return max;
+}
+
+RppStatus non_silent_region_detection_host_tensor(Rpp32f *srcPtr,
+                                                  RpptDescPtr srcDescPtr,
+                                                  Rpp32s *srcSizeTensor,
+                                                  Rpp32s *detectedIndexTensor,
+                                                  Rpp32s *detectionLengthTensor,
+                                                  Rpp32f *cutOffDBTensor,
+                                                  Rpp32s *windowLengthTensor,
+                                                  Rpp32f *referencePowerTensor,
+                                                  Rpp32s *resetIntervalTensor,
+                                                  bool *referenceMaxTensor)
+{
+#pragma omp parallel for num_threads(srcDescPtr->n)    
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+      Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->w;
+      Rpp32s srcSize = srcSizeTensor[batchCount];
+      Rpp32s windowLength = windowLengthTensor[batchCount];
+      Rpp32f referencePower = referencePowerTensor[batchCount];
+      Rpp32f cutOffDB = cutOffDBTensor[batchCount];
+      bool referenceMax = referenceMaxTensor[batchCount];
+
+      // set reset interval based on the user input 
+      Rpp32s resetInterval = resetIntervalTensor[batchCount];
+      resetInterval = (resetInterval == -1) ? srcSize : resetInterval;
+
+      // Calculate buffer size for mms array and allocate mms buffer
+      Rpp32s mmsBufferSize = srcSize - windowLength + 1;
+      std::vector<float> mmsBuffer;
+      mmsBuffer.reserve(mmsBufferSize);
+      
+      // Calculate moving mean square of input array and store in mms buffer
+      Rpp32f sumOfSquares = 0.0f;
+      Rpp32f meanFactor = 1.f / windowLength;
+      for (int windowBegin = 0; windowBegin <= srcSize - windowLength;) 
```

Comment:
If this loop has no increment, maybe use while loop

Reply:
Done